src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -264,24 +263,20 @@ void CARMACloudPlugin::Broadcast_TCMs()
 			for( auto itr = _not_ACK_TCMs->begin(); itr!=_not_ACK_TCMs->end(); ++itr )
 			{
 				string tcmv01_req_id_hex = itr->first;	
-				//Start recording time out period
-				if(!is_started_broadcasting)
+				auto cur_time = (std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch())).count();				
+				if (_tcm_broadcast_starting_time->count(tcmv01_req_id_hex) == 0)
 				{
-					start_time = (std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch())).count();
-				}else{
-					cur_time = (std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch())).count();
+					_tcm_broadcast_starting_time->insert({tcmv01_req_id_hex, cur_time});

```

Comment:
There can be multiple TCMs associated with a single TCM req ID and this is going to expire all of them which is probably fine in practice because they all will expire within msec apart but in theory, you should most likely use the TCM ID here instead as the key.  This probably applies to all checks.

Reply:
The time out should be for each TCR, right? After the vehicle sends a TCR and it receives the first TCM, we should count timer until it receives an ACK or time out.